app/models/voxelytics/VoxelyticsDAO.scala

```diff
@@ -257,358 +415,287 @@ class VoxelyticsDAO @Inject()(sqlClient: SQLClient)(implicit ec: ExecutionContex
-    for {
-      objectIdList <- run(q"""SELECT _id
-             FROM webknossos.voxelytics_tasks
-             WHERE _run = $runId AND name = $taskName
-             """.as[String])
-      objectId <- objectIdList.headOption
-    } yield ObjectId(objectId)
-
-  def getChunkIdByName(taskId: ObjectId, executionId: String, chunkName: String): Fox[ObjectId] =
-    for {
-      objectIdList <- run(q"""SELECT _id
-             FROM webknossos.voxelytics_chunks
-             WHERE _task = $taskId AND executionId = $executionId AND chunkName = $chunkName
-             """.as[String])
-      objectId <- objectIdList.headOption
-    } yield ObjectId(objectId)
-
-  def getArtifactIdByName(taskId: ObjectId, artifactName: String): Fox[ObjectId] =
-    for {
-      objectIdList <- run(q"""SELECT _id
-               FROM webknossos.voxelytics_artifacts
-               WHERE _task = $taskId AND name = $artifactName
-               """.as[String])
-      objectId <- objectIdList.headOption
-    } yield ObjectId(objectId)
-
-  def getChunkStatistics(taskId: ObjectId): Fox[List[ChunkStatisticsEntry]] = {
+  def getChunkStatistics(runIds: List[ObjectId],
+                         taskName: String,
+                         staleTimeout: FiniteDuration): Fox[List[ChunkStatisticsEntry]] = {
     for {
       r <- run(
         q"""
-          WITH latest_chunk_states AS (
-            SELECT DISTINCT ON (_chunk) _chunk, timestamp, state
-            FROM webknossos.voxelytics_chunkStateChangeEvents
-            ORDER BY _chunk, timestamp DESC
-          )
+        WITH latest_chunk_states AS (${latestChunkStatesQ(runIds, Some(taskName), staleTimeout)})
+        SELECT
+          exec.executionId AS executionId,
+          exec.countTotal AS countTotal,
+          exec.countFailed AS countFailed,
+          exec.countSkipped AS countSkipped,
+          exec.countComplete AS countComplete,
+          exec.countCancelled AS countCancelled,
+          times.beginTime AS beginTime,
+          times.endTime AS endTime,
+          times.wallTime AS wallTime,
+          profiling.max_memory AS max_memory,
+          profiling.median_memory AS median_memory,
+          profiling.stddev_memory AS stddev_memory,
+          profiling.max_cpuUser AS max_cpuUser,
+          profiling.median_cpuUser AS median_cpuUser,
+          profiling.stddev_cpuUser AS stddev_cpuUser,
+          profiling.max_cpuSystem AS max_cpuSystem,
+          profiling.median_cpuSystem AS median_cpuSystem,
+          profiling.stddev_cpuSystem AS stddev_cpuSystem,
+          durations.max_duration AS max_duration,
+          durations.median_duration AS median_duration,
+          durations.stddev_duration AS stddev_duration,
+          durations.sum_duration AS sum_duration
+        FROM ( -- Chunks grouped by executionId
+          SELECT
+            c.executionId,
+            COUNT(c._chunk) AS countTotal,
+            SUM(CASE WHEN c.state = 'FAILED' THEN 1 ELSE 0 END) AS countFailed,
```

Comment:
Should possibly use the enums as SqlValue? That way, usage search works here.

Reply:
good idea